services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/hanke/HankeAttachmentMigrator.kt

```diff
@@ -0,0 +1,93 @@
+package fi.hel.haitaton.hanke.attachment.hanke
+
+import fi.hel.haitaton.hanke.attachment.azure.Container.HANKE_LIITTEET
+import fi.hel.haitaton.hanke.attachment.common.AttachmentContent
+import fi.hel.haitaton.hanke.attachment.common.AttachmentValidator
+import fi.hel.haitaton.hanke.attachment.common.FileClient
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentContentEntity
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentContentRepository
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentEntity
+import fi.hel.haitaton.hanke.attachment.common.HankeAttachmentRepository
+import fi.hel.haitaton.hanke.attachment.common.MigrationResult
+import fi.hel.haitaton.hanke.attachment.common.UnMigratedHankeAttachment
+import java.util.UUID
+import mu.KotlinLogging
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+
+private val logger = KotlinLogging.logger {}
+
+@Component
+class HankeAttachmentMigrator(
+    private val fileClient: FileClient,
+    private val attachmentRepository: HankeAttachmentRepository,
+    private val contentRepository: HankeAttachmentContentRepository
+) {
+    /** Find an attachment that is not yet migrated, i.e. still has content in the db. */
+    @Transactional(readOnly = true)
+    fun unMigratedAttachment(): UnMigratedHankeAttachment? {
+        logAttachmentAmountToMigrate()
```

Comment:
This is not related to getting an unmigrated attachment. It could be called separately from the Scheduler.

Reply:
That would mean another dependency for the scheduler. I see this as part of the migration process so it should be in this class.

Reply:
This can be done without adding a dependency to the scheduler. `logAttachmentAmountToMigrate()` could be made public and called from the scheduler. Both classes are migration components, so which class handles which part is a pragmatic issue.
```kotlin
    fun scheduleMigrate() =
        lockService.doIfUnlocked(MIGRATE_HANKE_ATTACHMENT) {
            hankeAttachmentMigrator.logAttachmentAmountToMigrate()
            hankeAttachmentMigrator.findAttachmentWithDatabaseContent()?.let { migrate(it) }
                ?: logger.info { "No hanke attachments to migrate" }
        }
```
Then again, this is not a critical issue, especially since this code is short-lived.

Reply:
Oh, ok. I misunderstood that the unMigratedAttachment should be moved to the scheduler.

Reply:
The suggestion would be more logical I guess, but I think it would mean two different transactions. As said, this is a short lived code, so I'll leave it as it was.